salt/scripts.py

```diff
@@ -46,9 +46,24 @@ def _handle_signals(client, signum, sigframe):
         hardcrash = client.options.hard_crash
     except (AttributeError, KeyError):
         hardcrash = False
+
+    exit_msg = '\nExiting gracefully on Ctrl-c'
+    try:
+        jid = client.local_client.pub_data['jid']
```

Comment:
I'm concerned here because this signal handler is registered to all the CLI tools in `salt.scripts`. So, you could have salt-run killed and then you'd hit this and it would potentially stacktrace, for example.